src/util/version.ts

```diff
@@ -0,0 +1,18 @@
+export interface SemanticVersion {
+  major: i16,
+  minor: i16,
+  patch: i16
+}
+
+export function getSemanticVersion(_version: string): SemanticVersion {
+  let version = _version;
+  if (!version.match(/^([0-9]+)\.([0-9]+)\.([0-9]+)/)) {
+    version = "0.0.0";
+  }
+  const versionParts = version.split(".");
+  return {
+    major: parseInt(versionParts[0]),
+    minor: parseInt(versionParts[1]),
+    patch: parseInt(versionParts[2]),
+  };
+}
```

Comment:
Hi and thanks for this PR!
AssemblyScript also use self-bootstrap test and due to runtime doesn't support regexps yet will be great avoid it. Also I suggest inline this function. It will make returning object unnecessary

Reply:
So instead regexp you could use this:
```ts
 let major = 0, minor = 0, patch = 0;
 if (_version) { 
    const parts = _version.split(".");
    if (parts.length == 3) {
       major = max(parseInt(parts[0]) | 0, 0);
       minor = max(parseInt(parts[1]) | 0, 0);
       patch = max(parseInt(parts[2]) | 0, 0);
    }
 }
```

Reply:
For sure. Glad to jump on this great open source project!

Reply:
done @MaxGraey